lib/bolt/inventory.rb

```diff
@@ -243,5 +249,16 @@ def set_facts(tname, hash)
       end
     end
     private :set_facts
+
+    def add_node(grp, target, group_name)
+      grp.groups.each do |g|
+        add_node(g, target, group_name) if g.groups.any?
+        if g.name == group_name
+          g.nodes[target.uri.to_s] = { 'name' => target.uri.to_s }.merge(target.options)
+          break
```

Comment:
this control flow only stops the actual level in which the group was found. 
If we use `return true` instead we could short circuit more.